Change tooltip delay

Imran8125 opened the pull request:
<!-- YOU HAVE TO MODIFY THIS TEXT TO FIT YOUR PR. OTHERWISE, YOUR PR WILL BE CLOSED WITHOUT FURTHER COMMENT. -->

Description
This PR fixes tooltip delay from 2sec to 500ms and made changes to Base.css and MainTableTooltip

Changes
Updated Base.css to adjust tooltip delay settings.
Modified MainTableTooltip to align with these changes.
Why?
Ensures a smoother UX experience.
Closes #12649 


### Mandatory checks

- [x] I own the copyright of the code submitted and I licence it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if change is visible to the user)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

![Screenshot (5801)](https://github.com/user-attachments/assets/e2dc6aa4-833b-4d63-881c-ad6aa16d2e45)

subhramit commented on src/main/java/org/jabref/gui/maintable/MainTableTooltip.java:
```diff
@@ -23,7 +23,7 @@ public class MainTableTooltip extends Tooltip {
         this.preferences = preferences;
         this.preview = new PreviewViewer(dialogService, preferences, themeManager, taskExecutor);
-        this.setShowDelay(Duration.seconds(1));
+        this.setShowDelay(Duration.millis(500));
```
Please find out other places where we set the delay and make them uniform.
Please also check what the delay is for other common software (e.g. MS Word) and use that value.

Imran8125 replied:
ok

subhramit commented on CHANGELOG.md:
```diff
@@ -70,6 +71,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 - We fixed an issue where migration of "Search groups" would fail with an exception when the search query is invalid. [#12555](https://github.com/JabRef/jabref/issues/12555)
 - We fixed an issue where not all linked files from BibDesk in the field `bdsk-file-...` were parsed. [#12555](https://github.com/JabRef/jabref/issues/12555)
 - We fixed an issue where JabRef displayed an incorrect deletion notification when canceling entry deletion [#12645](https://github.com/JabRef/jabref/issues/12645)
+-Fixed tooltip delay from 2sec to 500ms in the main table[#12649](https://github.com/JabRef/jabref/issues/12649)
```
Please align the entry correctly (see the others). Also move it from to "Changed" section as it is not a bug fix. Add a full stop and a space as well before the issue number.

Imran8125 replied:
ok

Imran8125 commented on CHANGELOG.md:
```diff
@@ -48,6 +48,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 - We removed the obsolete Twitter link and added Mastodon and LinkedIn links in Help -> JabRef resources. [#12660](https://github.com/JabRef/jabref/issues/12660)
 - We improved journal abbreviation lookup with fuzzy matching to handle minor input errors and variations. [#12467](https://github.com/JabRef/jabref/issues/12467)
 
+
```
ok

Imran8125 commented:
1)I have updated the tooltip delay in Base.css, ensuring uniformity across all places.
2)I checked the tooltip delay in MS Word and other common software, and they also use 500ms. I have applied the same value in Base.css to ensure consistency.
3)I updated the CHANGELOG as per your suggestion
Let me know if any further adjustments are needed

subhramit commented:
> 1)I have updated the tooltip delay in Base.css, ensuring uniformity across all places.

If that is the case, why was a separate change needed at the maintable class?

Imran8125 commented:
> > 1)I have updated the tooltip delay in Base.css, ensuring uniformity across all places.
> 
> If that is the case, why was a separate change needed at the maintable class?

The Base.css change ensures uniform delay, but MainTable had a separate setting that needed adjustment for consistency

subhramit commented on src/main/java/org/jabref/gui/maintable/MainTableTooltip.java:
```diff
@@ -23,7 +23,7 @@ public class MainTableTooltip extends Tooltip {
         this.preferences = preferences;
         this.preview = new PreviewViewer(dialogService, preferences, themeManager, taskExecutor);
-        this.setShowDelay(Duration.seconds(1));
+        this.setShowDelay(Duration.millis(500));
```
I think this can be removed then? @Siedlerchr

Siedlerchr replied:
yes

koppor replied:
It is still in the code. WHY?

Imran8125 replied:
@koppor I have now removed it to ensure consistency. Let me know if any further changes are needed.

koppor reviewed (changes requested):
500ms is still to slow for me as impatient user.

That the tooltip suddenly disappears (without fade-out) puzzles me. Therefore the proposal to remove it.

koppor commented:
I updated the branch to bring in https://github.com/JabRef/jabref/pull/12736 here, too.
